click/testing.py

```diff
@@ -278,6 +279,14 @@ def invoke(self, cli, args=None, input=None, env=None,
                 if e.code != 0:
                     exception = e
                 exit_code = e.code
+                # Fix for issue #362: `SystemExit.code` sometimes contains
+                # the error message, not the exit code.
+                try:
+                    if exit_code != int(exit_code):
+                        raise ValueError
+                except ValueError:
```

Comment:
Here's the behavior of `sys.exit` for some corner cases:
```
$ python -c "import sys; sys.exit(1)"; echo $?
1
$ python -c "import sys; sys.exit(1.0)"; echo $?
1.0
1
$ python -c "import sys; sys.exit('1')"; echo $?
1
1
```
So integer float values are treated as non-integers.

So you were right after all, `isinstance(exit_code, int)` is the right choice. It's fixed now.